packages/x-ui-collections-bundle/src/graphql/Collection.ts

```diff
@@ -895,4 +920,32 @@ export abstract class Collection<T = null> {
   protected isUpdateModifier(element: any): element is UpdateFilter<T> {
     return element["$set"];
   }
+
+  /**
+   * Provide a Apollo.OptimisticResponse object based on the provided _id and document
+   */
+  public createOptimisticResponse(
+    _id: ObjectId | string,
+    document: Partial<T>,
+    operationName?: string
+  ) {
+    const stringId = _id instanceof ObjectId ? _id.toString() : _id;
+
+    // We can't be optimistic if we do not know the operationName
+    if (!operationName) return undefined;
+
+    const collectionName = this.getName();
+
+    // We can't be optimistic if we do not know the __typename
+    if (!collectionName) return undefined;
+
+    return {
+      // `${this.getName()}UpdateOne`
+      [operationName]: {
+        _id: stringId,
+        __typename: this.getTypeName(),
```

Comment:
Yep, we should also add this to blueprint collection generation to return the entity name so we don't have situations like "Goose" "Geese"